### PR TITLE
[HDInsight] - Update configurations

### PR DESCRIPTION
**Pull request by aim-for-better**
## Hydra PR Links

[Azure/hydra-specs-pr#1402: [HDInsight] - Update configurations](https://github.com/Azure/hydra-specs-pr/pull/1402)
## Rest Spec PR Links
[Azure/azure-rest-api-specs#5477: [HDInsight] - Update configuration related APIs ](https://github.com/Azure/azure-rest-api-specs/pull/5477)
## Note
Add 3 new SDK method groups:

1. IClusterOperations.ListConfigurationsAsync: Gets all configuration information for an HDI cluster.
2. IClusterOperations.GetGatewaySettingsAsync: Gets the gateway settings for the specified cluster.
3. IClusterOperations.UpdateGatewaySettingsAsync: Configures the gateway settings on the specified cluster.

### CODE REVIEW

**Review by idear1203 — APPROVED** [review]
LGTM.

**Comment by dsgouda** [comment]
@shahabhijeet PTAL

**Comment by dsgouda** [comment]
@aim-for-better Please join Azure org [here](https://aka.ms/azuregithub)

**Comment by aim-for-better** [comment]
> @aim-for-better Please join Azure org [here](https://aka.ms/azuregithub)
> Please link the Rest spec PR too

@dsgouda  The Rest spec PR Link has been updated. Please see the latest comment. Thanks.

**Comment by aim-for-better** [comment]
> > @aim-for-better Please join Azure org [here](https://aka.ms/azuregithub)
> > Please link the Rest spec PR too
> 
> @dsgouda The Rest spec PR Link has been updated. Please see the latest comment. Thanks.

And Hydra spec PR Link can be opened correctly now

**Comment by dsgouda** [comment]
@aim-for-better @idear1203 Why is this targeting the HyakNetCore branch?
@shahabhijeet PTAL

File: src/ResourceManagement/HDInsight/Microsoft.Azure.Management.HDInsight/Microsoft.Azure.Management.HDInsight.csproj [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (idear1203) [inline src/ResourceManagement/HDInsight/Microsoft.Azure.Management.HDInsight/Microsoft.Azure.Management.HDInsight.csproj]: Where does the file come from?

**Comment by idear1203** [comment]
>     <AssemblyOriginatorKeyFile>MSSharedLibKey.snk</AssemblyOriginatorKeyFile>

Is assembly sign necessary? Our Swagger based SDK don't have assembly sign. You can see from [here](https://github.com/Azure/azure-sdk-for-net/blob/master/src/SDKs/HDInsight/Management/Management.HDInsight/Management.HDInsight.csproj)

---
Refers to: src/ResourceManagement/HDInsight/Microsoft.Azure.Management.HDInsight/Management.HDInsight.csproj:15 in d21d960. [](commit_id = d21d960e582ccbdea0319449ef11067182a1fd1d, deletion_comment = False)

File: src/ResourceManagement/HDInsight/HDInsight.Tests/HDInsight.Tests.csproj [hunk 2]
```diff
@@ -5,6 +5,9 @@
     <AssemblyName>Microsoft.Azure.Management.HDInsight.Tests</AssemblyName>
     <Version>1.0.0</Version>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <SignAssembly>true</SignAssembly>
+    <DelaySign>true</DelaySign>
+    <AssemblyOriginatorKeyFile>..\Microsoft.Azure.Management.HDInsight\MSSharedLibKey.snk</AssemblyOriginatorKeyFile>
```
Comment (idear1203) [inline src/ResourceManagement/HDInsight/HDInsight.Tests/HDInsight.Tests.csproj]: >..\Microsoft.Azure.Management.HDInsight\MSSharedLibKey.snk [](start = 31, length = 58)

Looks a little bit strange to me that key file is referenced from another project.

Suggest two options:
1. Move a copy MSSharedLibKey.snk under the same folder with `HDInsight.Tests.csproj`, then change this line as `<AssemblyOriginatorKeyFile>MSSharedLibKey.snk</AssemblyOriginatorKeyFile>`
2. Change this line to `<AssemblyOriginatorKeyFile>..\..\..\..\tools\MSSharedLibKey.snk</AssemblyOriginatorKeyFile>`

File: src/ResourceManagement/HDInsight/HDInsight.Tests/ScenarioTests/ListClusterTests.cs [hunk 3]
```diff
@@ -69,7 +69,7 @@ public void TestListClustersInSubscription()
             }
         }
 
-        //[Fact]
+        [Fact(Skip = "This test case weill be skiped.")]
```
Comment (idear1203) [inline src/ResourceManagement/HDInsight/HDInsight.Tests/ScenarioTests/ListClusterTests.cs]: >weill [](start = 37, length = 5)

Typo

File: src/ResourceManagement/HDInsight/HDInsight.Tests/ScenarioTests/CreateClusterTests.cs [hunk 4]
```diff
@@ -59,12 +59,12 @@ public void TestPaasCreateGetDeleteCluster()
...
                 var result = client.Clusters.Delete(resourceGroup, dnsname);
-                Assert.Equal(result.StatusCode, HttpStatusCode.OK);
-                Assert.Equal(result.State, AsyncOperationState.Succeeded);
+                Assert.Equal(HttpStatusCode.OK, result.StatusCode);
+                Assert.Equal(AsyncOperationState.Succeeded, result.State);
             }
         }
 
-        //[Fact]
+        [Fact(Skip = "This test case will be skiped.")]
```
Comment (idear1203) [inline src/ResourceManagement/HDInsight/HDInsight.Tests/ScenarioTests/CreateClusterTests.cs]: >skiped [](start = 45, length = 6)

skipped

File: src/ResourceManagement/HDInsight/HDInsight.Tests/ScenarioTests/CreateClusterTests.cs [hunk 5]
```diff
@@ -177,17 +177,17 @@ public void TestCreateDuplicateCluster()
...
-                Assert.Equal(result.StatusCode, HttpStatusCode.OK);               
-                Assert.Equal(result.State, AsyncOperationState.Succeeded);
+                Assert.Equal( HttpStatusCode.OK, result.StatusCode);               
+                Assert.Equal(AsyncOperationState.Succeeded, result.State);
                 
             }
         }
 
-        //[Fact]
+        [Fact(Skip = "This teset case will be skiped.")]
```
Comment (idear1203) [inline src/ResourceManagement/HDInsight/HDInsight.Tests/ScenarioTests/CreateClusterTests.cs]: >teset [](start = 27, length = 5)

test

Comment (idear1203) [inline src/ResourceManagement/HDInsight/HDInsight.Tests/ScenarioTests/CreateClusterTests.cs]: >skiped [](start = 46, length = 6)

skipped

File: src/ResourceManagement/HDInsight/Microsoft.Azure.Management.HDInsight/Microsoft.Azure.Management.HDInsight.csproj (hunk 1 above)
Comment (aim-for-better) [reply src/ResourceManagement/HDInsight/Microsoft.Azure.Management.HDInsight/Microsoft.Azure.Management.HDInsight.csproj]: fixed

**Comment by aim-for-better** [comment]
> > ```
> > <AssemblyOriginatorKeyFile>MSSharedLibKey.snk</AssemblyOriginatorKeyFile>
> > ```
> 
> Is assembly sign necessary? Our Swagger based SDK don't have assembly sign. You can see from [here](https://github.com/Azure/azure-sdk-for-net/blob/master/src/SDKs/HDInsight/Management/Management.HDInsight/Management.HDInsight.csproj)
> 
> Refers to: src/ResourceManagement/HDInsight/Microsoft.Azure.Management.HDInsight/Management.HDInsight.csproj:15 in [d21d960](https://github.com/Azure/azure-sdk-for-net/commit/d21d960e582ccbdea0319449ef11067182a1fd1d). [](commit_id = [d21d960](https://github.com/Azure/azure-sdk-for-net/commit/d21d960e582ccbdea0319449ef11067182a1fd1d), deletion_comment = False)

yes, it is necessary. Otherwise, while building there is error.

File: src/ResourceManagement/HDInsight/HDInsight.Tests/ScenarioTests/CreateClusterTests.cs (hunk 5 above)
Comment (aim-for-better) [reply src/ResourceManagement/HDInsight/HDInsight.Tests/ScenarioTests/CreateClusterTests.cs]: fixed

File: src/ResourceManagement/HDInsight/HDInsight.Tests/ScenarioTests/CreateClusterTests.cs (hunk 4 above)
Comment (aim-for-better) [reply src/ResourceManagement/HDInsight/HDInsight.Tests/ScenarioTests/CreateClusterTests.cs]: fixed

File: src/ResourceManagement/HDInsight/HDInsight.Tests/ScenarioTests/ListClusterTests.cs (hunk 3 above)
Comment (aim-for-better) [reply src/ResourceManagement/HDInsight/HDInsight.Tests/ScenarioTests/ListClusterTests.cs]: fixed

File: src/ResourceManagement/HDInsight/HDInsight.Tests/HDInsight.Tests.csproj (hunk 2 above)
Comment (aim-for-better) [reply src/ResourceManagement/HDInsight/HDInsight.Tests/HDInsight.Tests.csproj]: modified based on 2 option

**Comment by aim-for-better** [comment]
>     <AssemblyOriginatorKeyFile>MSSharedLibKey.snk</AssemblyOriginatorKeyFile>

Yes it is necessary.  Otherwise, while building there is error

---
In reply to: [481169757](https://github.com/Azure/azure-sdk-for-net/pull/5685#issuecomment-481169757) [](ancestors = 481169757)

---
Refers to: src/ResourceManagement/HDInsight/Microsoft.Azure.Management.HDInsight/Management.HDInsight.csproj:15 in c204128. [](commit_id = c204128e8a02003b4d6c320054221c61de5ae2d1, deletion_comment = False)

**Review by idear1203 — APPROVED** [review]
LGTM.

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**